File: challenges/notification-dispatcher/go/starter/exercise.go
package exercise

import "refresher/tooling/go/contracts"

// Add the state required by the challenge contract.
type NotificationDispatcher struct{}

func NewNotificationDispatcher(store contracts.KeyValueStore, queue contracts.JobQueuePort,
	sender contracts.Sender, idempotencyTTLMS int64) *NotificationDispatcher {
	panic("Not implemented")
}

func (d *NotificationDispatcher) Submit(key, message string) string {
	panic("Not implemented")
}

func (d *NotificationDispatcher) ProcessOne() string {
	panic("Not implemented")
}

File: challenges/notification-dispatcher/php/starter/exercise.php
<?php
declare(strict_types=1);

namespace Refresher;

use Closure;
use LogicException;

final class NotificationDispatcher
{
    public function __construct(
        KeyValueStore $store,
        JobQueuePort $queue,
        Closure $sender,
        int $idempotencyTtlMs,
    ) {
        throw new LogicException("Not implemented");
    }

    public function submit(string $key, string $message): string
    {
        throw new LogicException("Not implemented");
    }

    public function processOne(): string
    {
        throw new LogicException("Not implemented");
    }
}
